src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py

```diff
@@ -847,6 +847,30 @@ def snapshot_info(self, snapname):
                                       "snapshot '{0}' does not exist".format(snapname))
             raise VolumeException(-e.args[0], e.args[1])
 
+    def check_dangling_clone_index(self, snapname):
```

Comment:
This entire function is repetitive. The `snapshot_info` calls the `get_pending_clones` which does the same checks. I suggest you to just add the symlink validation and removal in `get_pending_clones` (https://github.com/ceph/ceph/pull/55838/commits/cc99a18b16cf5ce332b2f4c3092f3235c909a86a#diff-2afc20b5874dc7ad850a970bcba2be7246f04c94d6ab0a44e2140f8491feaf44R746) which is a lot simple change.

Reply:
Updated!!